pkg/api/handler/default.go

```diff
@@ -562,6 +483,39 @@ func (m ArcDefaultHandler) validateBEEFTransaction(ctx context.Context, txValida
 	return nil
 }
 
+func (m ArcDefaultHandler) submitTransactions(ctx context.Context, txs []*bt.Tx, options *metamorph.TransactionOptions) ([]*metamorph.TransactionStatus, *api.ErrorFields) {
+	var submitStatuses []*metamorph.TransactionStatus
+
+	if len(txs) == 1 {
+		tx := txs[0]
+
+		// probably we could use SubmitTransactions() as well, but right now I don't want to create potential performance issues
```

Comment:
Noted, remove this comment :)

Reply:
Or actually, this comment may stay and I just realised that, if I'm not mistaken, we only use SubmitTransaction -> PutTransacion in this one place. So maybe, if we decide to use SubmitTransactionS here, we can remove the entire path SubmitTx -> PutTx for single transaction.

I would create a task for researching if that option is viable :)

Reply:
Come on! This comment explains the decisions made in the algorithm's implementation. If comments aren't meant for this, then what are they for?!

Reply:
You've forced me to do benchmarks which I didn't want to spend time on right now.
Benchmarks for methods `PutTransactions` and `PutTransaction`:
1. PutTransactions
```
goarch: amd64
pkg: github.com/bitcoin-sv/arc/internal/metamorph
cpu: AMD Ryzen 7 5800H with Radeon Graphics         
BenchmarkPutTransactions/txs-1-16         	  654842	      1681 ns/op	     792 B/op	      11 allocs/op
BenchmarkPutTransactions/txs-10-16        	  261242	      4585 ns/op	    3456 B/op	      29 allocs/op
BenchmarkPutTransactions/txs-100-16       	   46315	     26168 ns/op	   30193 B/op	     209 allocs/op
BenchmarkPutTransactions/txs-1000-16      	    5203	    236261 ns/op	  296689 B/op	    2009 allocs/op
PASS
ok  	github.com/bitcoin-sv/arc/internal/metamorph	5.177s
```
2. PutTransaction
```
pkg: github.com/bitcoin-sv/arc/internal/metamorph
cpu: AMD Ryzen 7 5800H with Radeon Graphics         
BenchmarkPutTransaction/txs-1-16         	 4224109	       282.2 ns/op	     288 B/op	       3 allocs/op
PASS
ok  	github.com/bitcoin-sv/arc/internal/metamorph	1.552s
```

We're sticking with the implementation for single transactions.

Reply:
Okay, looks like PutTransaction is around 6 times faster for a single transaction. That's surprisingly a lot! Maybe there are some possible improvements (in the future ofc) for batch submitting.

Epic benchmarks, thanks for doing it and sorry if you had to spend too much time on it 😅